Makefile.am

```diff
@@ -220,15 +193,13 @@ clean-local::
 maintainer-clean-local::
 	rm -rf dist/ node_modules/ package-lock.json
 
-node_modules/.stamp: package.json
-	@echo ''
-	@echo '    package.json changed'
-	@echo ''
-	@echo '    Please run:'
-	@echo ''
-	@echo '        tools/npm-install'
-	@echo ''
-	@false
+FORCE:
+
+V_NODE_MODULES = $(V_NODE_MODULES_$(V))
+V_NODE_MODULES_ = $(V_NODE_MODULES_$(AM_DEFAULT_VERBOSITY))
+V_NODE_MODULES_0 = @V=0
+package-lock.json: FORCE
+	$(V_NODE_MODULES) tools/node-modules make_package_lock_json
```

Comment:
I think we'll need to discuss this point face-to-face, since from my viewpoint, the thing you're complaining about is almost the entire benefit of this change.

Reply:
So you suggest we have two cases? (1) developer builds from git could involve internet-y things during `make`, and (2) builds from release tarballs can't ever change package.json and `make` will be guaranteed to work offline. I can certainly live with that. This PR does (1) now, and we need to check what happens in the (2) case -- i.e. `make dist`, unpack the tarball, touch package.json, and see what explodes, and if that error message needs to be improved somehow.

Reply:
I'll add some extra checks to make sure that this stuff really absolutely never runs from a tarball.